lib/zendesk_apps_support/package.rb

```diff
@@ -93,20 +202,64 @@ def requirements_json
       read_json('requirements.json')
     end
 
-    def translations(locale)
-      file_path = "translations/#{locale}.json"
-      file_path = "translations/en.json" unless file_exists?(file_path)
-      read_json(file_path, false)
+    def templates
+      @templates ||= begin
+                       templates_dir = File.join(@dir, 'templates')
+                       Dir["#{templates_dir}/*.hdbs"].inject({}) do |h, file|
+                         str = File.read(file)
+                         str.chomp!
+                         h[File.basename(file, File.extname(file))] = str
+                         h
+                       end
+                     end
+    end
+
+    def translations
+      @translations ||= begin
+        translation_dir = File.join(@dir, 'translations')
+        return {} unless File.directory?(translation_dir)
+
+        locale_path = "#{translation_dir}/#{self.default_locale}.json"
+        default_translations = process_translations(locale_path)
+        Dir["#{translation_dir}/*.json"].inject({}) do |h, tr|
+          locale              = File.basename(tr, File.extname(tr))
+          locale_translations = if locale == self.default_locale
+                                  default_translations
+                                else
+                                  default_translations.deep_merge(process_translations(tr))
+                                end
+          h[locale] = locale_translations
+          h
+        end
+      end
+    end
+
+    def translation_files
+      files.select { |f| f =~ /^translations\// }
+    end
+
+    def market_translations(locale)
+      result = translations[locale].try(:[], 'app') || {}
+      result.delete('name')
+      result.delete('description')
+      result.delete('long_description')
+      result.delete('installation_instructions')
+      result
+    end
+
+    def process_translations(locale_path)
+      translations = File.exist?(locale_path) ? JSON.parse(File.read(locale_path)) : {}
+      translations['app'].delete('package') if translations.has_key?('app')
+      remove_zendesk_keys(translations)
     end
 
     def app_translations(locale)
-      remove_zendesk_keys(translations(locale))
+      remove_zendesk_keys(translations[locale])
     end
 
-    def readified_js(app_name, app_id, asset_url_prefix, settings = {}, locale = 'en')
+    def readified_js(app_id, asset_url_prefix, locale = 'en')
```

Comment:
I wonder if we should use a hash here rather than relying on us knowing the argument order.